x-pack/plugins/index_lifecycle_management/public/application/sections/edit_policy/components/phases/shared_fields/data_tier_allocation_field/data_tier_allocation_field.tsx

```diff
@@ -142,7 +148,7 @@ export const DataTierAllocationField: FunctionComponent<Props> = ({ phase, descr
           phase={phase}
           nodes={nodesByAttributes}
           disableDataTierOption={Boolean(
-            isCloudEnabled && !hasDataNodeRoles && isUsingDeprecatedDataRoleConfig
+            isPreV8 && isCloudEnabled && isUsingDeprecatedDataRoleConfig
```

Comment:
Actually, I'm confused about this version check. Why is this logic concerned about which version we're on? We're concerned with the Cloud environment and the configuration of the cluster, so I don't see why version matters.

Reply:
I see your point, in this case the logic in Kibana does not have to be an exact mirror of the requirements matrix (added in the PR description) since "node types" (deprecated config) is not support in v8+ clusters. Will simplify the code!